File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.java
class Solution {
    public int solution(int[] nums) {
        boolean[] list = new boolean[200001];
        int answer = 0;

        for (int i = 0; i < nums.length; i++) {
            for (int j = 0; j < nums.length; j++) {
                if (!list[nums[i]]) {
                    list[nums[i]] = true;
                    answer++;
                }
            }
        }
        if (answer > nums.length / 2) {
            answer = nums.length / 2;
        }
        
        return answer;
    }
}

File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.cpp
#include <bits/stdc++.h>
using namespace std;

set<int> s;

int solution(vector<int> nums)
{
    int answer = nums.size() / 2;
    
    for(int i : nums){
        s.insert(i);
    }
    
    if(s.size() < answer) answer = s.size();
    
    
    return answer;
}